src/main/java/org/opensearch/geospatial/index/mapper/xypoint/XYPoint.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.mapper.xypoint;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+/**
+ * Represents a point in a 2-dimensional planar coordinate system with no range limitations
+ */
+@AllArgsConstructor
+@Getter
+@NoArgsConstructor
+public class XYPoint {
+    private double x;
```

Comment:
why double? Shouldn't this be float? Also, why not use Lucene's XY point ?

Reply:
Refactored the code as per our offline discussion